modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2122,13 +2122,40 @@ def to_arrow(result, op=frame._op, tables=[], frames=iter(input)):
 
         return result
 
-    def _build_index_cache(self):
-        """Materialize index and store it in the cache."""
+    def _compute_axis_labels_and_lengths(self, axis: int, partitions=None):
```

Comment:
the `._compute_axis_labels..()` method inherited from `PandasDataframe` didn't work well with HDK, so I had to re-implement it.